[Tanishq Sharma] iP

Tanishq4331 opened the pull request:
### Duke
> Don't quote me on this - but Duke is _**on point**_ :ok_hand:    -Martin Luther King Jr.

Getting overwhelmed by things you need to do? Fret not - Duke is here for you. It's,
- **here for you** (can't say this enough)
- smart
- understanding 
- supportive 


Here's how you summon `this` (Duke's preferred pronoun - although `this` doesn't mind being called Duke): 

1. download `this` from [here](https://www.youtube.com/watch?v=dQw4w9WgXcQ).
2. double-click `this`.
3. add your tasks.
4. let `this` manage your tasks for you.

You don't need features - all you need is Duke. But if you insist `this` also has features!!!
  - [x] Managing tasks
  - [x] Managing deadlines
  - [x] Managing events
  - [ ] Life advice ~~if you ask nicely~~

If you are a Java programmer, you can use Duke to practice Java too. Here's the `main` method:
```java
    public static void main(String[] args) {
        String filePath = System.getProperty("user.dir") + File.separator + "tasks.txt";
        new Duke(filePath).run();
    }
```

Commit:
Add Gradle support

Tanishq4331 committed:
Complete Level 1

Tanishq4331 committed:
Fix level 1

Tanishq4331 committed:
Complete Level 2

Tanishq4331 committed:
Complete Level 3

Tanishq4331 committed:
Complete Level 4

Tanishq4331 committed:
Semi-Automate testing

Tanishq4331 committed:
Handle Errors

Tanishq4331 committed:
Add ability to delete tasks

Tanishq4331 committed:
Add Level-7 functionality: Save

Tanishq4331 committed:
Add Level-8 functionality: Dates and Times

Tanishq4331 committed:
Merge branch 'branch-level-7'

Tanishq4331 committed:
Merge branch 'branch-level-8'

# Conflicts:
#	src/main/java/Event.java
#	src/main/java/Task.java
#	text-ui-test/ACTUAL.TXT

Tanishq4331 committed:
Fix merge conflicts

Tanishq4331 committed:
Use More OOP

Tanishq4331 committed:
Organize into Packages

Tanishq4331 committed:
Add JUnit Tests

Tanishq4331 committed:
Create a JAR File

Tanishq4331 committed:
Add JavaDocs

Tanishq4331 committed:
Follow the Coding Standard

Tanishq4331 committed:
Implement Find

Tanishq4331 committed:
Merge branch 'branch-Level-9'

Tanishq4331 committed:
Merge branch 'branch-A-CodingStandard'

# Conflicts:
#	src/main/java/duke/main/Parser.java

Tanishq4331 committed:
Merge branch 'branch-A-JavaDoc'

# Conflicts:
#	src/main/java/duke/main/Parser.java
#	src/main/java/duke/main/Ui.java
#	src/main/java/duke/task/Task.java
#	src/main/java/duke/task/TaskList.java

Tanishq4331 committed:
Use class method parseTime from Task instead of Parser

Tanishq4331 committed:
Merge branch 'master' into add-gradle-support

kevin9foong commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,65 @@
+
+        while (true) {
+            try {
+                boolean keepParsing = parser.parse(input);
```
Here's a minor nit, but boolean variable names should be named to sound like booleans. Perhaps something like `shouldKeepParsing` may be better?

Tanishq4331 replied:
You're right, it should follow the boolean naming scheme.

kevin9foong commented on src/main/java/duke/main/Parser.java:
```diff
@@ -0,0 +1,88 @@
+    private Task selectTask(String taskNumString) {
+        try {
+            int taskNum = Integer.parseInt(taskNumString);
+            Task inFocus = taskList.getTask(taskNum);
```
perhaps this variable `inFocus` could be changed to `taskInFocus` so that it'll be easier to remember what this variable name is when used later on.

Tanishq4331 replied:
Thanks for the suggestion. I'll make the change.

kevin9foong commented on src/main/java/duke/task/Event.java:
```diff
@@ -0,0 +1,92 @@
+import java.time.LocalDate;
+import java.util.Objects;
+
+public class Event extends Task {
```
it might be a good idea to add a header javadoc comment for public classes

Tanishq4331 replied:
Right, I totally missed the javadoc.

kevin9foong commented on src/main/java/duke/task/Task.java:
```diff
@@ -0,0 +1,98 @@
+     *
+     * @return String for storing the Task.
+     */
+    abstract public String storageString();
```
Based on the coding guidelines, this method name could be further improved to use a verb as the name instead. Maybe something like `generateStorageString`?

Tanishq4331 replied:
I must have missed this. Thanks for pointing out the violation.

kevin9foong commented on src/main/java/duke/task/TaskList.java:
```diff
@@ -0,0 +1,157 @@
+            throw new DukeException("\tYou haven't specified any keywords\n");
+        }
+        String[] keywords = query.split(" ");
+        List<Task> matches = new ArrayList<>();
```
I like how you used plural form for variable names which represent multiple values.

kevin9foong reviewed (commented):
Hi Tarnishq, I've been assigned to make a PR review on code quality guidelines for your code. Hopefully my comments will be useful! 😃

Tanishq4331 committed:
Use Gradle

Tanishq4331 committed:
Merge branch 'add-gradle-support'

Tanishq4331 committed:
Edit gitignore

Tanishq4331 committed:
Use CheckStyle

Tanishq4331 committed:
Improve variable naming

Tanishq4331 committed:
Configure GUI for Duke

Tanishq4331 committed:
Design Ui Layout; Move Ui setup from Duke.java to Ui.java

Tanishq4331 committed:
Use FXML for Ui

IrvingHe000 commented on src/main/java/duke/main/Parser.java:
```diff
@@ -0,0 +1,88 @@
+     * @param input String from user.
+     * @return true if input is not "bye", else false.
+     */
+    public boolean parse(String input) {
```
Should the name of the method be named like a boolean? Because this method returns a boolean variable :)

Tanishq4331 replied:
Good catch! I've made the change.

IrvingHe000 commented on src/main/java/duke/task/TaskList.java:
```diff
@@ -0,0 +1,157 @@
+     *
+     * @return String summary of TaskList.
+     */
+    public String taskSummary() {
```
Perhaps it's better to change the method name to getTaskSummary since it's a method?

Tanishq4331 replied:
Thanks for pointing this out. I totally missed it.

IrvingHe000 commented on src/main/java/duke/task/TaskList.java:
```diff
@@ -0,0 +1,157 @@
+     * @param query for matching Tasks.
+     * @return List of matching Tasks.
+     */
+    public List<Task> find(String query) {
```
Maybe can change find to a more meaningful name?

IrvingHe000 commented on src/main/java/duke/main/Storage.java:
```diff
@@ -0,0 +1,91 @@
+     *
+     * @return Loaded TaskList.
+     */
+    public TaskList load() {
```
Maybe is better to change to a more meaningful name such as loadTaskList?

IrvingHe000 commented on src/main/java/duke/main/Storage.java:
```diff
@@ -0,0 +1,91 @@
+     *
+     * @param tasklist that is written to Storage
+     */
+    public void write(TaskList tasklist) {
```
The same as above.

Tanishq4331 replied:
Thanks for the suggestions. I'll make these method names more descriptive.

IrvingHe000 reviewed (commented):
Overall, I think your code is well written and easy to read 👍. Just some nits related to naming standards can be improved :)

Tanishq4331 committed:
Enable basic Duke responses.

Tanishq4331 committed:
Fix Task display formatting.

Tanishq4331 committed:
Add close on bye command.

Tanishq4331 committed:
Add missing javadocs.

Tanishq4331 committed:
Merge branch 'branch-Level-10'

Tanishq4331 committed:
Update some method names.

Tanishq4331 committed:
Update build.gradle mainClassName to duke.Launcher

Gradle uses duke.Duke to build the JAR file.

This no longer works since Duke.Main has been removed during the transition from command-line to GUI.

Since the program now runs from the duke.Launcher class, build.gradle needs to be updated for the JAR file to open the GUI instead.

Tanishq4331 committed:
Add Assertions

Tanishq4331 committed:
Improve Code Quality

The parse method in Parser handles most of the parsing and executing. Apply the SLAP principle to Parser to better distinguish between the two and increase readability.

 Other refactorings to meet the code quality standards.

Tanishq4331 committed:
Merge pull request #1 from Tanishq4331/A-Assertions

A-Assertions

Tanishq4331 committed:
Merge branch 'master' into A-CodeQuality

Tanishq4331 committed:
Merge pull request #2 from Tanishq4331/A-CodeQuality

Improve Code Quality

Tanishq4331 committed:
Add basic tagging functionality

Changes the structure of a task to include a list of tags.

Adds additional command "tag" to enable the user to tag a task.

Formatted Strings for storing and retrieving tasks have been updated to include tags.

Show tags when displaying tasks.

Tanishq4331 committed:
Update tests to work with tags

Tanishq4331 committed:
Merge branch 'C-Tagging'

Tanishq4331 committed:
Refactor parts of Parser into Executor

Tanishq4331 committed:
Fix executor bug in Duke

Tanishq4331 committed:
Change GUI font to Courier New to add macOS compatibility.

Tanishq4331 committed:
Fix write to Storage bug.

Tanishq4331 committed:
Fix javadocs in Duke

Tanishq4331 committed:
Improve UI Text formatting

Tanishq4331 committed:
Improve GUI

Tanishq4331 committed:
Set theme jekyll-theme-slate

Tanishq4331 committed:
Add Ui screenshot

Tanishq4331 committed:
Make Duke window larger

Tanishq4331 committed:
Fix scrollbar

Tanishq4331 committed:
Update gitignore

Tanishq4331 committed:
Merge branch 'master' of github.com:Tanishq4331/ip

Tanishq4331 committed:
Add user guide

Tanishq4331 committed:
Update UG

Tanishq4331 committed:
Update UG

Tanishq4331 committed:
Reduce MainWindow height for better compatibility wih smaller viewports

Tanishq4331 committed:
Merge branch 'master' of github.com:Tanishq4331/ip

Tanishq4331 committed:
Fix empty description bug

Tanishq4331 committed:
Fix method headers
